remote.rb

```diff
@@ -3,24 +3,139 @@
+      index += 1
+    end
+
+    if element.css('td font tt').size == 7
+      comission_element = element.css('td font tt')[5].text.strip.gsub(/\"/,'')
+      total_element     = element.css('td font tt')[6].text.strip.gsub(/\"/,'')
+    elsif element.css('td font tt').size == 6
+      comission_element = element.css('td font tt')[4].text.strip.gsub(/\"/,'')
+      total_element     = element.css('td font tt')[5].text.strip.gsub(/\"/,'')
+    end
+
+    hash.push(
+      trans_date:   Date.strptime(element.css('td font tt')[0].text.strip.gsub(/\"/,''), '%d/%m/%y').to_json,
+      details:      element.css('td font tt')[1].text.strip.gsub(/\"/,''),
+      amount:       element.css('td font tt')[2].text.strip.gsub(/\"/,''),
+      currency:     element.css('td font tt')[3].text.strip.gsub(/\"/,''),
+      post_date:    post_d,
+      comission:    comission_element,
+      total_amount: total_element
+    )
+  end
+  return hash
+end
+
+def other_trans_array (array, hash)
+  array.each do |element|
+    units    = element.css('td font tt')
+    tds      = units.map { |td| td.text.strip.gsub(/\"/,'') }
+    hash.push(
+      post_date:      Date.strptime(tds[0], '%d/%m/%y').to_json,
+      trans_date:     Date.strptime(tds[1], '%d/%m/%y').to_json,
+      details:        tds[2],
+      trans_amount:   tds[3],
+      trans_currency: tds[4],
+      acc_amount:     tds[5],
+      acc_currency:   tds[6]
+      )
+  end
+  return hash
+end
+
+def data (array, item, elements_arr, transactions, other_transactions)
+  array.push(  
+    name:               item.css('.contract-number').text,
+    balance:            item.css('.owwb-cs-slide-list-amount-value').text.to_f,
+    currency:           item.css('.owwb-cs-slide-list-amount-currency').text,
+    nature:             elements_arr[elements_arr.index("Produsul")+1],
+    transactions:       transactions,
+    other_transactions: other_transactions
+  )
+end
+
+def trans_page (arr_1, arr_2, arr_3, page)
+  page.css('tbody tr').each do |tr| 
+    if tr.text.strip.match(/^\d{2}\/\d{2}\/\d{2}\s*[A-z]\S*/)
+      arr_1 << tr
+    elsif tr.text.strip.match(/^\d{2}\/\d{2}\/\d{2}\s{1,5}\d{2}\/\d{2}\/\d{2}/)
+      arr_2 << tr
+    elsif tr.text.strip.match(/^\d{2}\/\d{2}\/\d{2}$/)
+      arr_3 << tr.css('td font tt').text.strip.gsub(/\"/,'')
+    end   
+  end 
+end
+
+data_arr    = []
+itter       = 1
+index       = 0
+post_d      = nil
+login       = "********"
+password    = "********"
+day         = "owwb-ws-calendar-current-month-day"
+from_date   = "USER_PROPERTY_DATE_FROM_DATEPICKER_SWITCH"
+nav_prev    = "owwb-ws-calendar-nav-prev"
+trans_entry = "owwb_cs_submit_emulator form_owwb_ws_entryAccountDoing-"
+confirm     = "owwb-cs-default-button-bg"
+back_main   = "form_GO_TO_PARENT_SECTION"
+
+browser = Watir::Browser.new(:firefox)
+browser.goto 'https://wb.micb.md/frontend/auth/userlogin?execution=e2s1'
+browser.text_field(:name => "Login").when_present.set(login)
```

Comment:
you can leave it this way, but you should implement gets. anyone should be able to get his data from his web banking. not just me